lib/modules/home/screens/search.dart

```diff
@@ -22,14 +63,66 @@ class _SearchScreenState extends State<SearchScreen> {
     );
   }
 
+  void updateQuery() {
+    timer.cancel();
+    setState(() {
+      query = formQuery(algolia.index(sort.index), textController.text);
+      pagingController.refresh();
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        title: const Text('Search'),
+        title: TextField(
+          controller: textController,
+          autofocus: true,
+          decoration: const InputDecoration.collapsed(
+            hintText: 'Search',
+          ),
+        ),
+        actions: [
+          if (textController.text.isNotEmpty)
+            IconButton(
```

Comment:
I think it's better to move this into the `InputDecoration`'s `suffixIcon` property. The only drawback would be having to remove `const`

Reply:
Are you sure about that? I tried it myself like this:
```dart
decoration: InputDecoration(
  hintText: 'Search',
  suffixIcon: IconButton(
    onPressed: textController.clear,
    icon: const Icon(Icons.clear_rounded),
  ),
),
```
and it worked even better (the button took the color of the text field). I also tried it in the [InputDecoration.suffixIcon playground](https://api.flutter.dev/flutter/material/InputDecoration/suffixIcon.html) and it was just fine.

Reply:
Their setup on the playground is different. I want to dynamically hide/show the button depending on whether the input text is empty. For that UX I use such code:
```dart
 decoration: InputDecoration(
            hintText: 'Search',
            border: InputBorder.none,
            suffix: textController.text.isEmpty
                ? null
                : IconButton(
                    onPressed: textController.clear,
                    icon: const Icon(Icons.clear_rounded),
                  ),
          ),
```
This code makes the field jump a few pixels on the button appearance. A similar bug appears when you put the button into the `suffix` property.

Reply:
I don't know what's the difference between `suffix` and `suffixIcon`, but in my experience leaving the button there even if there is no text is a common thing to do. If you want, you can disable it by setting `onPressed` to `null`, but I see no reason to remove it altogether. Another solution could be to add some dummy widget to take up that space; like, just leave a `Container(width: 0)` instead of `null`.
But in general, I believe that semantics are important, and having the button inside the text field is more important than hiding it when the text is empty, because this way it will also inherit some styling from `TextField` automatically.